py/picca/pk1d/prep_pk1d.py

```diff
@@ -94,71 +98,99 @@ def exp_diff_desi(hdul, mask_targetid):
+    file: fitsio.fitslib.FITS
+    Header Data Unit List opened by fitsio.
+
+    mask_targetid: array of int
+    Targetids to select for calculating the exposure differences.
+
+    method_alpha: str
+    Method to compute alpha. Defaults to "desi_array".
 
-    mask targetid: array of int
-    Targetids to select for calculating the exp differences
+    use_only_even: bool
+    Flag to indicate whether to use only even-number exposures. Defaults to False.
 
     Return
     ------
     diff: array of float
     The difference between exposures
     """
-    argsort = np.flip(np.argsort(hdul["TEFF_LYA"][mask_targetid][:]))
-    flux = hdul["FL"][mask_targetid][argsort, :]
-    ivar = hdul["IV"][mask_targetid][argsort, :]
-    teff_lya = hdul["TEFF_LYA"][mask_targetid][argsort]
+    argsort = np.flip(np.argsort(np.mean(file["IV"][mask_targetid], axis=1)))
+
+    teff_lya = file["TEFF_LYA"][mask_targetid][argsort]
+    flux = file["FL"][mask_targetid][argsort, :]
+    ivar = file["IV"][mask_targetid][argsort, :]
 
     num_exp = len(flux)
     if num_exp < 2:
-        userprint("DBG : not enough exposures for diff, spectra rejected")
+        print("Not enough exposures for diff, spectra rejected")
         return None
-    flux_total_odd = np.zeros(flux.shape[1])
-    ivar_total_odd = np.zeros(flux.shape[1])
-    flux_total_even = np.zeros(flux.shape[1])
-    ivar_total_even = np.zeros(flux.shape[1])
-    teff_even = 0
-    teff_odd = 0
-    teff_total = np.sum(teff_lya)
-    teff_last = teff_lya[-1]
-    for index_exp in range(2 * (num_exp // 2)):
-        flexp = flux[index_exp]
-        ivexp = ivar[index_exp]
-        teff_lya_exp = teff_lya[index_exp]
-        if index_exp % 2 == 1:
-            flux_total_odd += flexp * ivexp
-            ivar_total_odd += ivexp
-            teff_odd += teff_lya_exp
-        else:
-            flux_total_even += flexp * ivexp
-            ivar_total_even += ivexp
-            teff_even += teff_lya_exp
-
-    w = ivar_total_odd > 0
-    flux_total_odd[w] /= ivar_total_odd[w]
-    w = ivar_total_even > 0
-    flux_total_even[w] /= ivar_total_even[w]
-
-    alpha = 1
-    if num_exp % 2 == 1:
-        # Ignasi: these variables were unsused so I commented them out
-        # n_even = (num_exp - 1) // 2
-        # alpha_n_old = np.sqrt(4. * n_even * (num_exp - n_even)) / num_exp
-        # alpha_N = np.sqrt(4.*t_even*(t_exp-t_even))/t_exp
-        # alpha_c_new = np.sqrt((teff_total - teff_last) / teff_total)
-        alpha_n_new = np.sqrt(
-            (teff_total - teff_last) * (teff_total + teff_last)) / teff_total
-        alpha = alpha_n_new
-    diff = 0.5 * (flux_total_even - flux_total_odd) * alpha
-
+    if use_only_even:
+        if num_exp % 2 == 1:
+            print("Odd number of exposures discarded")
+            return None
+
+    time_even = 0
+    time_odd = 0
+    time_exp = np.sum(teff_lya)
+
+    even_inds = slice(0, 2 * (num_exp // 2), 2)
+    odd_inds = slice(1, 2 * (num_exp // 2), 2)
```

Comment:
note that as written this will automatically discard the last exposure if an odd total number of exposures is there, so potentially we wouldn't need the `use_only_even` above at all (except that that currently does remove objects not exposures)...
If we actually wanted to include the case of odd-exposures fully, we'd need to use `even_inds = slice(0, 2*(num_exp//2) + 1, 2)`